src/AutoRest.CSharp/Mgmt/Decorator/ReferenceClassFinder.cs

```diff
@@ -23,6 +25,57 @@ public class ReferenceClassFinder
         internal const string SerializationCtorAttributeName = "SerializationConstructorAttribute";
         internal const string ReferenceTypeAttributeName = "ReferenceTypeAttribute";
 
+        public record PropertyMetadata(string SerializedName, bool Required)
+        {
+            public PropertyMetadata(string serializedName) : this(serializedName, true)
+            {
+            }
+        }
+
+        private static readonly Dictionary<Type, Dictionary<string, PropertyMetadata>> _referenceTypesPropertyMetadata = new()
+        {
+            [typeof(ResourceData)] = new()
+            {
+                ["Id"] = new PropertyMetadata("id"),
+                ["Name"] = new PropertyMetadata("name"),
+                ["ResourceType"] = new PropertyMetadata("type"),
+                ["SystemData"] = new PropertyMetadata("systemData", false),
+            },
+            [typeof(TrackedResourceData)] = new()
+            {
+                ["Id"] = new PropertyMetadata("id"),
+                ["Name"] = new PropertyMetadata("name"),
+                ["ResourceType"] = new PropertyMetadata("type"),
+                ["Location"] = new PropertyMetadata("location"),
+                ["Tags"] = new PropertyMetadata("tags", false),
+                ["SystemData"] = new PropertyMetadata("systemData", false)
+            },
+            [typeof(ManagedServiceIdentity)] = new()
+            {
+                ["PrincipalId"] = new PropertyMetadata("principalId"),
+                ["TenantId"] = new PropertyMetadata("tenantId"),
+                ["ManagedServiceIdentityType"] = new PropertyMetadata("type"),
+                ["UserAssignedIdentities"] = new PropertyMetadata("userAssignedIdentities"),
+            },
+            [typeof(SystemAssignedServiceIdentity)] = new()
+            {
+                ["PrincipalId"] = new PropertyMetadata("principalId"),
+                ["TenantId"] = new PropertyMetadata("tenantId"),
+                ["SystemAssignedServiceIdentityType"] = new PropertyMetadata("type"),
```

Comment:
Only type is required. 

You may consider make default to be optional as most properties in a `PropertyReferenceType` model is optional? Then we need to set all `ReferenceType` proeprties explicitly since the default for them should be required and you've already done that. Don't forget that we have a `TypeReferenceType` `ReponseError` in Azure.Core.

Reply:
Since most of the things are just following the rules, and I am tired to add all of them here, (it is hard to maintain if we have a really long list), I am planning:
1. make the required default to false in PropertyMetadata, and assign it to true in some special cases.
2. if the mapping of a model is so normal (just propertyName: ToCamelCase(propertyName)), I will not include that in this list. I will add a function to construct this mapping from a type. Whether a property is required is fetched from its public constructor.